src/napari_spatialdata/_view.py

```diff
@@ -322,6 +292,49 @@ def channel_changed(event: Event) -> None:
         self.model.events.adata.connect(self._on_layer_update)
         self.model.events.color_by.connect(self._change_color_by)
 
+    def _channel_changed(self, event: Event) -> None:
+        layer = self.model.layer
+        is_image = isinstance(layer, Image)
+
+        has_sdata = layer is not None and layer.metadata.get("sdata") is not None
+        has_adata = layer is not None and layer.metadata.get("adata") is not None
+
+        # has_adata is added so we see the channels in the view widget under vars
+        if layer is None or not is_image or not has_sdata or not has_adata:
```

Comment:
maybe it should look like 
```suggestion
        if layer is None or not is_image or not has_sdata or not has_adata or image.rgb:
```